polaris/benchmark/_base.py

```diff
@@ -310,6 +294,36 @@ def _compute_checksum(dataset, target_cols, input_cols, split, metrics):
         checksum = hash_fn.hexdigest()
         return checksum
 
+    def verify_checksum(self, md5sum: Optional[str] = None):
+        """
+        Recomputes the checksum and verifies whether it matches the stored checksum.
+        """
+        if md5sum is None:
+            md5sum = self._md5sum
+        if md5sum is None:
+            raise RuntimeError(
+                "No checksum to verify against. Specify either the md5sum parameter or "
+                "store the checksum in the benchmark._md5sum attribute."
+            )
+
+        # Temporarily reset
+        # Calling self.md5sum will recompute the checksum and set it again
+        self._md5sum = None
+        if self.md5sum != md5sum:
+            raise PolarisChecksumError(
+                f"The specified checksum {md5sum} does not match the computed checksum {self.md5sum}"
+            )
+
+    @computed_field
+    @property
+    def md5sum(self) -> Optional[str]:
+        """Lazily compute the checksum once needed."""
+        if self._md5sum is None:
+            self._md5sum = self._compute_checksum(
```

Comment:
Good point! This was a static method before because it was called from within a class method (i.e., a Pydantic `@model_validator`). That's no longer the case!